aiida/manage/tests/__init__.py

```diff
@@ -449,17 +449,32 @@ def aiida_profile():
 
 
 def get_test_backend_name():
-    """ Read name of database backend from environment variable.
+    """ Read name of database backend from environment variable or the specified test profile.
 
-    Reads database backend ('django' or 'sqlalchemy') from 'AIIDA_TEST_BACKEND' environment variable.
+    Reads database backend ('django' or 'sqlalchemy') from 'AIIDA_TEST_BACKEND' environment variable,
+    or the backend configured for the 'AIIDA_TEST_PROFILE'.
     Defaults to django backend.
 
     :returns: content of environment variable or `BACKEND_DJANGO`
     :raises: ValueError if unknown backend name detected.
+    :raises: ValueError if both 'AIIDA_TEST_BACKEND' and 'AIIDA_TEST_PROFILE' are set, and the two
+        backends do not match.
     """
-    backend_env = os.environ.get('AIIDA_TEST_BACKEND', BACKEND_DJANGO)
-    if backend_env in (BACKEND_DJANGO, BACKEND_SQLA):
-        return backend_env
+    test_profile_name = get_test_profile_name()
+    backend_env = os.environ.get('AIIDA_TEST_BACKEND', None)
+    if test_profile_name is not None:
+        backend_profile = configuration.load_config().get_profile(test_profile_name).database_backend
```

Comment:
Is this the right way to access the profile information, or is there a better way?

Reply:
Almost, I think `get_config()` (rather than `load_config()`) would be the appropriate function here

Reply:
Fixed!